torch/_dynamo/__init__.py

```diff
@@ -156,14 +162,96 @@ def mark_dynamic(t, index):
     4) Attempts to trace this function will explicitly raise. As such, all calls to mark_dynamic must be made
     before torch.compile.
 
+    """
+    mark_dynamic_constrain(t, index, min=None, max=None)
+
+
+@forbid_in_graph
+def mark_dynamic_constrain(
```

Comment:
OK, with the docblock, I recommend just folding this into `mark_dynamic`, let's not have two APIs for this.

Reply:
Sure, I am okay with that.